saleor/graphql/attribute/types.py

```diff
@@ -656,3 +655,791 @@ class AttributeValueInput(BaseInputObjectType):
+        description="List of assigned product variant references.",
+        required=True,
+        limit=PositiveInt(
+            description=(
+                "Maximum number of referenced product variants to return. "
+                f"Default is {DEFAULT_NESTED_LIST_LIMIT}."
+            ),
+            default_value=DEFAULT_NESTED_LIST_LIMIT,
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = (
+            "Represents multi product variant reference attribute." + ADDED_IN_322
+        )
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[ChannelContext[product_models.ProductVariant]]]:
+        if not root.values:
+            return Promise.resolve([])
+
+        channel_slug = root.attribute.channel_slug
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        def _wrap_with_channel_context(
+            variants: list[product_models.ProductVariant],
+        ) -> list[ChannelContext[product_models.ProductVariant]]:
+            if not variants:
+                return []
+            return [
+                ChannelContext(node=variant, channel_slug=channel_slug)
+                for variant in variants
+            ]
+
+        return (
+            ProductVariantByIdLoader(info.context)
+            .load_many([value.reference_variant_id for value in attr_values])
+            .then(_wrap_with_channel_context)
+        )
+
+
+class AssignedMultiCategoryReferenceAttribute(BaseObjectType):
+    value = NonNullList(
+        "saleor.graphql.product.types.Category",
+        description="List of assigned category references.",
+        required=True,
+        limit=PositiveInt(
+            description=(
+                "Maximum number of referenced categories to return. "
+                f"Default is {DEFAULT_NESTED_LIST_LIMIT}."
+            ),
+            default_value=DEFAULT_NESTED_LIST_LIMIT,
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents multi category reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[product_models.Category]]:
+        if not root.values:
+            return Promise.resolve([])
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        return CategoryByIdLoader(info.context).load_many(
+            [value.reference_category_id for value in attr_values]
+        )
+
+
+class AssignedMultiCollectionReferenceAttribute(BaseObjectType):
+    value = NonNullList(
```

Comment:
In this field, Saleor returns multiple values. 
```suggestion
    values = NonNullList(
```